src/registry.rs

```diff
@@ -83,47 +124,47 @@ impl Registry {
         self.r.write().unwrap().unregister(c)
     }
 
-    pub fn write_pb<T: Write>(&self, _: &mut T) -> Result<()> {
-        Ok(())
+    fn gather(&self) -> Vec<proto::MetricFamily> {
+        self.r.read().unwrap().gather()
     }
 
-    pub fn write_text<T: Write>(&self, _: &mut T) -> Result<()> {
-        Ok(())
-    }
-}
+    pub fn scrap(&self, writer: &mut Write, encoder: &Encoder) -> Result<usize> {
```

Comment:
As I said before, remove this function, only support gather + encoder like go client.

For case 1, I don't know what you mean. For every gather, like hyper `/metrics` handler, why can't we create a encoder for every request? 
For case 2, why we have two different encoder traits? If we support writer + encoder, so does it writes first and then encodes or vice verse?